solvers/fista_automatic_restart.py

```diff
@@ -0,0 +1,94 @@
+from benchopt import BaseSolver, safe_import_context
+
+with safe_import_context() as import_ctx:
+    import numpy as np
+    from scipy import sparse
+
+
+class Solver(BaseSolver):
+    name = 'Python-FISTA_automatic_restart'
+    stopping_strategy = "iteration"
+
+    references = [
+        'J.-F. Aujol, Ch. Dossal, H. Labarrière, A. Rondepierre, '
+        '"FISTA restart using an automatic estimation of the'
+        'growth parameter", HAL preprint : hal-03153525v4, '
+        'A. Beck and M. Teboulle,'
+        '"A fast iterative shrinkage-thresholding algorithm for'
+        ' linear inverse problems", SIAM J. Imaging Sci., '
+        'vol. 2, no. 1, pp. 183-202 (2009)'
+    ]
+
+    def skip(self, X, y, lmbd, fit_intercept):
+        if fit_intercept:
+            return True, f"{self.name} does not handle fit_intercept"
+        return False, None
+
+    def set_objective(self, X, y, lmbd, fit_intercept):
+        self.X, self.y, self.lmbd = X, y, lmbd
+        self.fit_intercept = fit_intercept
+
+    def run(self, n_iter):
+        L = self.compute_lipschitz_constant()
+        n_features = self.X.shape[1]
+        w = np.zeros(n_features)
+        z = np.zeros(n_features)
+        C = 6.38
+        k_tab = np.zeros(int(n_iter/(2*C))+2)
+        F_tab = np.zeros(int(n_iter/(2*C))+2)
+        i_glob = 0
+        i_int = 0
+        counter = 0
```

Comment:
can you try to have more explicit names like `outer_iteration`, `inner_iteration` etc?